project_1/src/main/java/app/service/impl/BookServiceImpl.java

```diff
@@ -62,9 +62,52 @@ public Integer countByName(String bookName) {
         }
     }
 
-    public List<BookInfo> randomBooks() {
+    @Override
+    public List<BookInfo> getPanel(int maxResult) {
+        try {
+            List<Book> panels = bookDAO.getPanel(maxResult);
+            if (panels != null){
+                List<BookInfo> panelInfo = new ArrayList<>();
+                for (Book book : panels){
+                    panelInfo.add(new BookInfo(book));
+                }
+                return panelInfo;
+            }
+            return Collections.emptyList();
```

Comment:
chị đề xuất code sửa từ line 69 -> 76 thành 
 if (panels == null){
     return Collections.emptyList();
}

List<BookInfo> panelInfo = new ArrayList<>();
for (Book book : panels){
          panelInfo.add(new BookInfo(book));
}
return panelInfo;